typescript/overview.md

```diff
@@ -53,27 +53,42 @@ One of the main advantages of TypeScript is that you can make code more type
 safe, so that what would be syntactically valid JavaScript becomes TypeScript
 with warnings about being "unsafe".
 
-In Deno we handle TypeScript in two major ways. We can type check TypeScript,
-the default, or you can opt into skipping that checking using the `--no-check`
-flag. For example if you had a program you wanted to run, normally you would do
-something like this:
+You can type-check your code (without executing it) using the following command:
 
-```
-deno run --allow-net my_server.ts
+```shell
+deno check module.ts
 ```
 
-But if you wanted to skip the type checking, you would do something like this:
+Type checking can take a significant amount of time, especially if you are
+working on a code base where you are making a lot of changes. We have tried to
+optimise the type checking, but it still comes at a cost. **Therefore, by
+default, TypeScript modules are not type-checked before they are executed.**
 
+```shell
+deno run module.ts
 ```
-deno run --allow-net --no-check my_server.ts
+
+When using the command above, Deno will simply transpile the module before
+executing it, ignoring any potential type-related issues. In order to perform a
+type-check of the module before execution occurs, the `--check` argument must be
+used with `deno run`:
+
+```shell
+deno run --check module.ts
 ```
 
-Type checking can take a significant amount of time, especially if you are
-working on a code base where you are making a lot of changes. We have tried to
-optimise the type checking, but it still comes at a cost. If you just want to
-hack at some code, or if you are working in an IDE which is type checking your
-code as you author it, using `--no-check` can certainly speed up the process of
-running TypeScript in Deno.
+While `tsc` will (by default) still emit JavaScript when encountering diagnostic
```

Comment:
Note: This sentence might be inaccurate now because of this change https://github.com/denoland/deno/pull/15118 ?

Reply:
@kt3k Thanks for reviewing.

I think it's setting up a juxtaposition between `tsc` (the unmodified TypeScript compiler binary) and Deno, and is creating a frame of reference for people who come from experience with vanilla TypeScript compilation behavior (at least that's how I read it).

If you're questioning the following part — the description of Deno's behavior when using `deno run --check` — I just tested and can confirm that it's accurate.

<details>
<summary>Here's the minimal test I performed:</summary>

```
% deno --version
deno 1.25.0 (release, x86_64-apple-darwin)
v8 10.6.194.5
typescript 4.7.4

% cat bad-types.ts 
function logUpperCase(...strings: string[]): void {
  console.log(...strings);
}

logUpperCase("It", "is", true);

% deno run --check bad-types.ts
Check file:///Users/deno/bad-types.ts
error: TS2345 [ERROR]: Argument of type 'boolean' is not assignable to parameter of type 'string'.
logUpperCase("It", "is", true);
                         ~~~~
    at file:///Users/deno/bad-types.ts:5:26

% echo $?
1

% cat bad-types-expected.ts 
function logUpperCase(...strings: string[]): void {
  console.log(...strings);
}

// @ts-expect-error
logUpperCase("It", "is", true);

% deno run --check bad-types-expected.ts
It is true

% echo $?
0

```
</details>

Reply:
I think the main point of the paragraph is ok as is.

I thought that we shouldn't use the word `tsc` here because we don't use `tsc` (but use `swc`) for transpiling typescripts internally now since https://github.com/denoland/deno/pull/15118

Maybe it's better not to mention `tsc` or `swc` at all here.
```suggestion
While Deno's internal compiler will (by default) still emit JavaScript when encountering diagnostic
```

Reply:
@kt3k Thanks for clarifying. I could be wrong (I haven't recently reviewed that part of the CLI source), but I feel confident that Deno does still use a modified version of `tsc` for _type-checking_ (which is what this passage is about). Maybe @kitsonk can weigh in here as well? I'm using the GitHub app on mobile at the moment and can't see the file history to be sure, but I suspect he might have been the original author of that statement.